application/models/services/Quotas.php

```diff
@@ -384,4 +392,212 @@ public function checkActionPermissions($action)
+                    if ($oQuota->qlimit === 0) {
+                        // Always add the quota if qlimit is 0
+                        $aMatchedQuotas[] = $oQuota->getViewArray();
+                    } else {
+                        $iCompleted = $oQuota->completeCount;
+                        if (!is_null($iCompleted) && ($iCompleted >= $oQuota->qlimit)) {
+                            // This remove invalid quota and not completed
+                            $aMatchedQuotas[] = $oQuota->getViewArray();
+                        }
+                    }
+                }
+            }
+        }
+        if ($return) {
+            return $aMatchedQuotas;
+        }
+        if (empty($aMatchedQuotas)) {
+            return;
+        }
+        // Now we have all the information we need about the quotas and their status.
+        // We need to construct the page and do all needed action
+        $aSurveyInfo = getSurveyInfo($surveyid, $_SESSION['survey_' . $surveyid]['s_lang']);
+        $sClientToken = $_SESSION['survey_' . $surveyid]['token'] ?? "";
+        // $redata for templatereplace
+        $aDataReplacement = [
+            'thissurvey'  => $aSurveyInfo,
+            'clienttoken' => $sClientToken,
+            'token'       => $sClientToken,
+        ];
+        // We take only the first matched quota, no need for each
+        $aMatchedQuota = $aMatchedQuotas[0];
+        // If a token is used then mark the token as completed, do it before event : this allow plugin to update token information
+        $event = new PluginEvent('afterSurveyQuota');
+        $event->set('surveyId', $surveyid);
+        $event->set('responseId', $_SESSION['survey_' . $surveyid]['srid']); // We always have a responseId
+        $event->set('aMatchedQuotas', $aMatchedQuotas); // Give all the matched quota : the first is the active
+        App()->getPluginManager()->dispatchEvent($event);
+        $blocks = [];
+        foreach ($event->getAllContent() as $blockData) {
+            /* @var $blockData PluginEventContent */
+            $blocks[] = CHtml::tag('div', ['id' => $blockData->getCssId(), 'class' => $blockData->getCssClass()],
+                $blockData->getContent());
+        }
+        // Allow plugin to update message, url, url description and action
+        $sMessage = $event->get('message', $aMatchedQuota['quotals_message']);
+        $sUrl = $event->get('url', $aMatchedQuota['quotals_url']);
+        $sUrlDescription = $event->get('urldescrip', $aMatchedQuota['quotals_urldescrip']);
+        $sAction = $event->get('action', $aMatchedQuota['action']);
+        /* Tag if we close or not the survey */
+        $closeSurvey = ($sAction == "1" || App()->getRequest()->getPost('move') == 'confirmquota');
+        $sAutoloadUrl = $event->get('autoloadurl', $aMatchedQuota['autoload_url']);
+        // Doing the action and show the page
+        if ($sAction == Quota::TERMINATE_AFTER_ALL_VISIBLE_QUOTA_QUESTION && $sClientToken) {
```

Comment:
`in_array($sAction, [Quota::TERMINATE_AFTER_ALL_VISIBLE_QUOTA_QUESTION,Quota::TERMINATE_AFTER_ALL_QUOTA_QUESTIONSSTION,Quota::TERMINATE_AFTER_ALL_GROUPS])`

or `$sAction != Quota::SOFT_TERMINATE_AFTER_ALL_VISIBLE_QUOTA_QUESTION`

Reply:
This one must be fixed, no real issue found on other comment.